adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpAdapterSaslAuthenticatorFactory.java

```diff
@@ -334,7 +334,7 @@ public boolean succeeded() {
                         try {
                             return certValidator.validate(
                                     Collections.singletonList(deviceCert),
-                                    tenant.getTrustAnchor()).map(ok -> tenant);
+                                    tenant.getTrustAnchors().get(0)).map(ok -> tenant);
```

Comment:
I would rather not commit this change as it seems kind of arbitrary, which certificate will be used as the trust anchor. Instead, I think that you should first extend the `X509CertificateChainValidator` interface with a method that accepts a list of TrustAnchors instead of just a single TrustAnchor. Please put this change into a separate commit to make it easier to see which changes are related to each other. Then we should use all the trust anchors contained in the `TenantObject` when performing the validation here.

Reply:
ok, i will do that 👍

Reply:
@sophokles73 patch updated